### PR TITLE
docs: document lack of config file names

### PR DESCRIPTION
**Pull request by Zamiell**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I had a very confusing time reading the ESLint docs today so I think this extra warning section will really help ESLint users who are migrating.

#### Is there anything you'd like reviewers to focus on?

No.

### CODE REVIEW

File: docs/src/use/configure/configuration-files-new.md [hunk 1]
```diff
@@ -667,7 +679,7 @@ When ESLint is run on the command line, it first checks the current working dire
 You can prevent this search for `eslint.config.js` by setting the `ESLINT_USE_FLAT_CONFIG` environment variable to `true` and using the `-c` or `--config` option on the command line to specify an alternate configuration file, such as:
 
 ```shell
-ESLINT_USE_FLAT_CONFIG=true npx eslint -c some-other-file.js **/*.js
+ESLINT_USE_FLAT_CONFIG=true npx eslint --config some-other-file.js **/*.js
```
Comment (Zamiell) [inline docs/src/use/configure/configuration-files-new.md:682]: While sort of unrelated to the warning that I added elsewhere, I think that using the full flag here makes the docs more clear.

File: docs/src/use/configure/configuration-files-new.md [hunk 2]
```diff
@@ -58,6 +58,18 @@ module.exports = (async () => {
...
+::: warning
+
+Unlike other tools in the JavaScript ecosystem, ESLint does not automatically look for a config file with a name of `eslint.config.mjs` or `eslint.config.cjs`. The ESLint team [does not want to support these file extensions](https://github.com/eslint/eslint/issues/16580) because we want to keep things simple: users are expected to use the same format for their ESLint files that the rest of their project uses, as defined in the `type` field of the project's `package.json` file.
+
+If you want to force a different kind of ESLint config, then name your config file accordingly and use the `-c` or `--config` flag when running ESLint. For example, if you want to have an ESM ESLint config in a CommonJS project, name the config file `eslint.config.mjs` and then run ESLint with:
+
+```shell
+ESLINT_USE_FLAT_CONFIG=true npx eslint --config eslint.config.mjs **/*.js
+```
+
```
Comment (nzakas) [inline docs/src/use/configure/configuration-files-new.md:70]: This is a bit aggressive, plus it completely breaks the layout. I'm fine with adding a warning but let's keep it simple.
```suggestion
ESLint only automatically looks for a config file named `eslint.config.js` and does not look for `eslint.config.cjs` or `eslint.config.mjs`. If you'd like to specify a different config filename than the default, use the `--config` command line option.
```

**Review by nzakas — CHANGES_REQUESTED** [review]

File: docs/src/use/configure/configuration-files-new.md (hunk 2 above)
Comment (Zamiell) [reply docs/src/use/configure/configuration-files-new.md]: lgtm

**Review by nzakas — APPROVED** [review]

**Comment by nzakas** [comment]
LGTM. Thanks.

**Pull request merged**